dbt/adapters/redshift/impl.py

```diff
@@ -171,26 +95,9 @@ def drop_relation(self, relation, model_name=None):
 
         https://docs.aws.amazon.com/redshift/latest/dg/r_DROP_TABLE.html
         """
-        global drop_lock
-
-        to_return = None
-
-        with drop_lock:
-
-            connection = self.get_connection(model_name)
-
-            if connection.transaction_open:
-                self.commit(connection)
-
-            self.begin(connection.name)
-
-            to_return = super(RedshiftAdapter, self).drop_relation(
-                relation, model_name)
-
-            self.commit(connection)
-            self.begin(connection.name)
-
-            return to_return
+        with self.connections.fresh_transaction(model_name):
```

Comment:
relevant: https://github.com/fishtown-analytics/dbt/issues/697

It looks like you're just using a brand new connection here, right? I _think_ that might be ok, but I'm curious what your thinking was around this change.

Reply:
This is functionally the same, I just extracted the transaction management into a context manager. It's not getting a new connection unless the named connection never existed. That context manager just triggers a (maybe commit) + begin, yields, and triggers a commit + begin again. Hence `fresh_transaction`.

I saw #697 and decided I'm not qualified on that front.